### PR TITLE
return selected file from org-roam-insert

### PR DESCRIPTION
**Pull request by d12frosted**
###### Motivation for this change

This is useful in scenarios when you want to automatically do something with the
context where the link was inserted. For example, if you are linking someone in
the task, you can add a special tag for this person.

```emacs-lisp
(defun +org-notes-insert ()
  "Insert a link to the note."
  (interactive)
  (when-let* ((res (org-roam-insert))
              (path (plist-get res :path))
              (title (plist-get res :title))
              (roam-tags (+seq-flatten
                          (+seq-flatten
                           (org-roam-db-query [:select tags
                                               :from tags
                                               :where (= file $s1)]
                                              path)))))
    (when (seq-contains-p roam-tags "People")
      (save-excursion
        (ignore-errors
          (org-back-to-heading)
          (let ((tags (org-get-tags)))
            (org-set-tags (seq-uniq (cons (+org-notes--title-to-tag title) tags)))))))))

(defun +org-notes--title-to-tag (title)
  "Convert TITLE to tag."
  (concat "@" (s-replace " " "" title)))
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ### Features
 - [#814] Implement `org-roam-insert-immediate`
+- [#839] Return selected file from `org-roam-insert`
```
Comment (d12frosted) [inline CHANGELOG.md:9]: Should I use a link to this pull request?

File: org-roam.el [hunk 2]
```diff
@@ -1378,7 +1379,8 @@ If DESCRIPTION is provided, use this as the link label.  See
                                                                :link-description link-description
                                                                :capture-fn 'org-roam-insert))
         (org-roam--with-template-error 'org-roam-capture-templates
-          (org-roam-capture--capture))))))
+          (org-roam-capture--capture))))
+    res))
```
Comment (d12frosted) [inline org-roam.el:1383]: if the user didn't select existing note, the value will be `nil`. We could return `title-with-tags`, but the problem is that `org-roam-insert` finishes before capture process is done. So it's totally fine to have `nil`.

**Comment by d12frosted** [comment]
Usage example:

auto-tag

**Comment by zaeph** [comment]
Looks interesting.  Where do you insert your `when` block, though?  Did you patch it in directly in `org-roam-insert`?  We could implement a hook for you to do this more easily.

**Comment by d12frosted** [comment]
No, I just have a wrapper for `org-roam-insert` - `+org-notes-insert` that I use instead of the former.

```emacs-lisp
(defun +org-notes-insert ()
  "Insert a link to the note."
  (interactive)
  (when-let* ((res (org-roam-insert))
              (path (plist-get res :path))
              (title (plist-get res :title))
              (roam-tags (+seq-flatten
                          (+seq-flatten
                           (org-roam-db-query [:select tags
                                               :from tags
                                               :where (= file $s1)]
                                              path)))))
    (when (seq-contains-p roam-tags "People")
      (save-excursion
        (org-back-to-heading)
        (let ((tags (org-get-tags)))
          (org-set-tags (seq-uniq (cons (+org-notes--title-to-tag title) tags))))))))
```

**Comment by d12frosted** [comment]
I think the hook could work as well. Especially with the newly captured files, which would be handy I think.

**Comment by zaeph** [comment]
Definitely a candidate for a hook.  I'll review your PR later today, and I'll probably create another one adding a hook

**Comment by d12frosted** [comment]
@zaeph thank you :) feel free to close this PR in favour of something more generic. I went the simplest path ;)

**Comment by zaeph** [comment]
Your code looks all right: I'd probably just rename `res` to something a little more meaningful.  If I find any problem with the code, I'll just push to your branch.  Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (jethrokuan) [reply CHANGELOG.md:9]: Yup, that would be good!

**Comment by d12frosted** [comment]
I've added a link to this PR to changelog. And also fixed the conflict with master ;) Thanks for the review!

**Comment by d12frosted** [comment]
Thanks for merging! ❤️

**Pull request merged**